feature/user_profile/src/main/java/org/mifos/mobile/feature/user_profile/screens/UserProfileScreen.kt

```diff
@@ -0,0 +1,324 @@
+                                client = uiState.client,
+                                isOnline = isOnline,
+                                bitmap = userBitmap!!,
+                                home = navigateBack
+                            )
+
+                        }
+
+                        is UserDetailState.ShowUserImage -> {
+                            userBitmap = getUserImage(uiState.image, context)
+                            userBitmap.let {
+                                UserProfileContent(
+                                    changePassword = changePassword,
+                                    client = client,
+                                    isOnline = isOnline,
+                                    bitmap = it!!,
+                                    home = navigateBack
+                                )
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    )
+}
+
+@Composable
+fun UserProfileContent(
+    changePassword: () -> Unit,
+    client : Client,
+    isOnline: Boolean,
+    bitmap: Bitmap,
+    home: () -> Unit
+) {
+    val context = LocalContext.current
+
+    val userName = nullFieldCheck(context.getString(R.string.username), client.displayName, context )
+    val accountNumber = nullFieldCheck(context.getString(R.string.account_number), client.accountNo, context)
+    val activationDate = nullFieldCheck(
+        context.getString(R.string.activation_date),
+        DateHelper.getDateAsString(client.activationDate),
+        context
+    )
+    val officeName = nullFieldCheck(context.getString(R.string.office_name), client.officeName, context)
+    val clientType = nullFieldCheck(context.getString(R.string.client_type), client.clientType?.name, context)
+    val groups = nullFieldCheck(context.getString(R.string.groups), getGroups(client.groups, context), context)
+    val clientClassification = client.clientClassification?.name ?: "-"
+    val phoneNumber = nullFieldCheck(context.getString(R.string.phone_number), client.mobileNo, context)
+    val dob = if (client.dobDate.size != 3) {
+        context.getString(R.string.no_dob_found)
+    } else {
+        DateHelper.getDateAsString(client.dobDate)
+    }
+    val gender = nullFieldCheck(context.getString(R.string.gender), client.gender?.name, context)
+    val userDetails = UserDetails(
+        userName = userName,
+        accountNumber = accountNumber,
+        activationDate = activationDate,
+        officeName = officeName,
+        clientType = clientType,
+        groups = groups,
+        clientClassification = clientClassification,
+        phoneNumber = phoneNumber,
+        dob = dob,
+        gender = gender
+    )
+
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .verticalScroll(rememberScrollState())
+    ) {
+        UserProfileTopBar(home = home, text = R.string.user_details)
+
+        if (!isOnline) {
```

Comment:
remove.  We don't check for network connectivity unless we're doing network-related operations.